service/src/main/java/bio/terra/workspace/service/workspace/flight/cloud/aws/DeleteWorkspaceApplicationSecurityGroupsStep.java

```diff
@@ -0,0 +1,79 @@
+package bio.terra.workspace.service.workspace.flight.cloud.aws;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.common.utils.AwsUtils;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.iam.SamService;
+import bio.terra.workspace.service.workspace.AwsCloudContextService;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.regions.Region;
+
+public class DeleteWorkspaceApplicationSecurityGroupsStep implements Step {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(DeleteWorkspaceApplicationSecurityGroupsStep.class);
+  private final CrlService crlService;
+  private final SamService samService;
+  private final AwsCloudContextService awsCloudContextService;
+  private final UUID workspaceUuid;
+
+  public DeleteWorkspaceApplicationSecurityGroupsStep(
+      CrlService crlService,
+      AwsCloudContextService awsCloudContextService,
+      SamService samService,
+      UUID workspaceUuid) {
```

Comment:
Same here - this can be fetched from flightmap

Reply:
See above.